magento/pwa-studio
# PWA-2571: Regression fix disabled TTL support in BrowserPersistence

- Reverted ttl support for getItem



## Description

Problem
A user auth bug was fixed in v11.0.0 regression, but it appears to have removed support for an important feature for TTL support of entries. While this may have solved the bug, this utility is public API, and anyone else who was using it expecting items to expire now have large stale caches for their shoppers.

Introduced by: https://github.com/magento/pwa-studio/pull/3306/files#diff-44b7b17ad1392c88b1e3411c80f7378782e6a568d280b68a48349061293b604cL51-L54

Acceptance Criteria
TTL is restored as supported in BrowserPersistence
No regressions related to the user auth bug this intended to fix


## Related Issue







Closes #PWA-2571

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps





Test Execution
Verify https://github.com/magento/pwa-studio/pull/3306. still works
Verify TTL support works as expected. Browser stale cache should expire.
Verify above 2 scenarios in all supported browsers and devices.
Verify apollo does not store any sensitive data after user session logout
Verify above scenario in both DEfault. and French store
Cypress Test coverage -
Changes should be covered in unit test level. No cypress unless unit test does not support this kind of scenario to test.

#### Test scenario(s) for direct fix/feature





#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
